keps/sig-instrumentation/1748-pod-resource-metrics/README.md

```diff
@@ -0,0 +1,497 @@
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Expose new metrics
+
+The scheduler will expose an optional set of metrics series that capture the resources a pod is considered to be "consuming" from a scheduling perspective at the current time.
+
+The series reported will be consistent with `kube_node_status_allocatable` to simplify queries that compare used capacity and available capacity.
+
+* `kube_pod_resource_requests` contains series for the consumed requested resources for a pod at a given time.
+  * The value is the quantity of the limit in `spec.containers.resources.requests`
+  * It has the following labels:
+    * `pod` - the name of the pod
+    * `namespace` - the namespace of the pod
+    * `resource` - the name of the resource as described in `spec.containers.resources.requests`
+    * `units` - the units of the resource type as inferred from the resource name in `spec.containers.requests`, such as `bytes`, `cores`. Empty if this is a unitless resource.
+* `kube_pod_resource_limits` contains series for the consumed limit resources for a pod at a given time.
+  * The value is the quantity of the limit in `spec.containers.resources.limits`
+  * It has the following labels:
+    * `pod` - the name of the pod
+    * `namespace` - the namespace of the pod
+    * `resource` - the name of the resource as described in `spec.containers.resources.limits`
+    * `units` - the units of the resource type as inferred from the resource name in `spec.containers.limits`, such as `bytes`, `cores`. Empty if this is a unitless resource.
+* `kube_pod_resource_requests_lifecycle` contains series for the consumed requested resources for a pod at a given time when the pod is in the provided lifecycle phase. All pods are in one of the lifecycle phases, but series will only be returned for the current phase.
```

Comment:
Based on our previous discussion, I think I know what you want to achieve with this metric, but that isn't "consumed requested resources".  In terms of access to resources, a pod's `Initializing` phase is identical to the `running` phase, as their total access to resources is dictated by pod-level cgroups, which don't change between phases.

Reply:
yeah, the data an admin gets is "where am i unbalanced in terms of initializing vs post-initialization", or "when resource reservations are changed (later), how did usage compare to both reservation, effective reservation, and actual consumption"